mint/mint.go

```diff
@@ -231,6 +249,55 @@ type VerificationParams struct {
 	TargetService string
 }
 
+// Generates a pseudo-random secret.
+func (m *Mint) generatePseudoRandomSecret(id *lsat.Identifier) ([lsat.SecretSize]byte, error) {
+	var ret [lsat.SecretSize]byte
+
+	mac := hmac.New(sha256.New, []byte(*m.cfg.SeedForPseudoRandomness))
+	_, err := mac.Write(id.TokenID[:])
+	if err != nil {
+		return [lsat.SecretSize]byte{}, err
+	}
+
+	result := mac.Sum(nil)
+	if len(result) < lsat.SecretSize {
+		return [lsat.SecretSize]byte{}, ErrIncompatibleHmacFuncUsed
+	}
+
+	copy(ret[0:], result[0:lsat.SecretSize])
+
+	return ret, nil
+}
+
+// Either returns the secret from SecretStore or creates a pseudo-random one.
+func (m *Mint) determineSecret(ctx context.Context, params *VerificationParams, id *lsat.Identifier) ([lsat.SecretSize]byte, error) {
+	if m.cfg.SeedForPseudoRandomness == nil {
```

Comment:
nit: flip the condition around. Meaning `if ... != nil { return m.generatePseudoRandomSecret() }` as the branch is the path less often taken (or non-default path).